PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,470 @@
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization0p83To1p24_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization0p83To1p24_->Eval(pt);
+    statuncty = parametrization0p83To1p24_->GetParError(2);
+  } else if (std::abs(eta) < 1.4) {
+    if (parametrization1p24To1p4_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization1p24To1p4_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization1p24To1p4_->Eval(pt);
+    statuncty = parametrization1p24To1p4_->GetParError(2);
+  } else if (std::abs(eta) < 1.6) {
+    if (parametrization1p4To1p6_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization1p4To1p6_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization1p4To1p6_->Eval(pt);
+    statuncty = parametrization1p4To1p6_->GetParError(2);
+  } else if (std::abs(eta) < 1.8) {
+    if (parametrization1p6To1p8_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization1p6To1p8_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization1p6To1p8_->Eval(pt);
+    statuncty = parametrization1p6To1p8_->GetParError(2);
+  } else if (std::abs(eta) < 2.1) {
+    if (parametrization1p8To2p1_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization1p8To2p1_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization1p8To2p1_->Eval(pt);
+    statuncty = parametrization1p8To2p1_->GetParError(2);
+  } else if (std::abs(eta) < 2.25) {
+    if (parametrization2p1To2p25_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization2p1To2p25_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization2p1To2p25_->Eval(pt);
+    statuncty = parametrization2p1To2p25_->GetParError(2);
+  } else if (std::abs(eta) < 2.4) {
+    if (parametrization2p25To2p4_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization2p25To2p4_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization2p25To2p4_->Eval(pt);
+    statuncty = parametrization2p25To2p4_->GetParError(2);
+  } else
+    return 0.;
```

Comment:
no `LogWarning` here?

Reply:
I have added a warning about the muon being out of acceptance for the weight parametrization.